atomicapp/providers/openshift.py

```diff
@@ -36,91 +37,302 @@ class OpenShiftProvider(Provider):
+            logger.error(msg)
+            raise ProviderFailedException(msg)
+
+    def _kind_to_resource(self, kind):
+        """
+        Converts kind to resource name. It is same logics
+        as in k8s.io/kubernetes/pkg/api/meta/restmapper.go (func KindToResource)
+        Example:
+            Pod -> pods
+            Policy - > policies
+            BuildConfig - > buildconfigs
+
+        Args:
+            kind (str): Kind of the object
+
+        Returns:
+            Resource name (str) (kind in plural form)
+        """
+        singular = kind.lower()
+        if singular.endswith("status"):
+            plural = singular + "es"
+        else:
+            if singular[-1] == "s":
+                plural = singular
+            elif singular[-1] == "y":
+                plural = singular.rstrip("y") + "ies"
+            else:
+                plural = singular + "s"
+        return plural
+
+    def _get_url(self, namespace, kind, name=None):
+        """
+        Some kinds/resources are managed by OpensShift and some by Kubernetes.
+        Here we compose right url (Kubernets or OpenShift) for given kind.
+        If resource is managed by Kubernetes or OpenShift is determined by
+        self.kapi_resources/self.oapi_resources lists
+        Example:
+            For namespace=project1, kind=DeploymentConfig, name=dc1 result
+            would be http://example.com:8443/oapi/v1/namespaces/project1/deploymentconfigs/dc1
+
+        Args:
+            namespace (str): Kubernetes namespace or Openshift project name
+            kind (str): kind of the object
+            name (str): object name if modifying or deleting specific object (optional)
+
+        Returns:
+            Full url (str) for given kind, namespace and name
+        """
+        url = None
+
+        resource = self._kind_to_resource(kind)
+
+        if resource in self.oapi_resources:
+            url = self.openshift_api
+        elif resource in self.kapi_resources:
+            url = self.kubernetes_api
+
+        url = urljoin(url, "namespaces/%s/%s/" % (namespace, resource))
+
+        if name:
+            url = urljoin(url, name)
+
+        url = urljoin(url, "?access_token={}".format(self.access_token))
+        logger.debug("url: %s", url)
+        return url
+
+    def _parse_kubeconf(self, f):
+        """"
+        Parse kubectl config file
+
+        Args:
+            f (string): path to configuration file (e.g. ./kube/config)
+
+        Returns:
+            tuple with server url, oauth token and namespace
+            (url, token, namespace)
+        """
+
+        url = None
+        token = None
+        namespace = None
+
+        logger.debug("Parsing %s", f)
+
+        with open(f, 'r') as fp:
+            kubecfg = anymarkup.parse(fp.read())
+
+        current_context = kubecfg["current-context"]
+
+        logger.debug("current context: %s", current_context)
+
+        context = None
```

Comment:
The code is beatifully written but I do think if we had an example of what the file looks like somewhere here in the comments it would make the code easier to read